python/ray/data/_internal/stats.py

```diff
@@ -280,6 +280,11 @@ def __init__(self, max_stats=1000):
             description="Seconds user thread is blocked by iter_batches()",
             tag_keys=iter_tag_keys,
         )
+        self.iter_time_to_first_batch_s = Gauge(
+            "data_iter_time_to_first_batch_seconds",
+            description="Seconds from starting iteration until the first batch is ready. This includes the dataset pipeline warmup time.",
```

Comment:
Cumulative time spent waiting for the first batch after starting iteration. This includes multiple first-batch wait times if dataset execution restarts across epochs.